lib/fs_utils/generate.js

```diff
@@ -85,7 +84,7 @@ const concat = (files, path, definitionFn, autoRequire, config) => {
   const processor = (file) => {
     root.add(file.node);
     const data = file.node.isIdentity ? file.data : file.source;
-    if (isJs && data.trim().substr(-1) !== semi) root.add(semi);
+    if (isJs && !data.trim().endsWith(';')) root.add(';');
```

Comment:
won't this allocate strings over and over? This may be a method that's executed lots and lots of times every sec.